modules/uhsg_sisu/src/Services/StudyRightsService.php

```diff
@@ -0,0 +1,368 @@
+                code
+                groupId
+                name {
+                  fi
+                  sv
+                  en
+                }
+              }
+            }
+          }
+        }
+      }',
+    ];
+
+    try {
+      $data = $this->sisuService->apiRequest($query);
+
+      // Save to static storage.
+      $this->studyRightsData[$oodiId] = $data;
+      return $this->studyRightsData[$oodiId];
+
+    }
+    catch (GuzzleException $e) {
+      $this->guzzleErrorLog($e);
+    }
+    catch (\Exception $e) {
+      $variables = Error::decodeException($e);
+      $this->log('%type: @message in %function (line %line of %file) @backtrace_string.', $variables, RfcLogLevel::ERROR);
+    }
+
+    $this->log('StudyRightsService encountered an unknown error when fetching StudyRights with query@studyrights', ['@studyrights' => $query], RfcLogLevel::ERROR);
+    return FALSE;
+  }
+
+  /**
+   * get all active studyrights for person.
+   *
+   * @param string $oodiId
+   *   Student Number.
+   *
+   * @return array|null
+   *   array of studyright obj or NULL.
+   */
+  public function getActiveStudyRights($oodiId) {
+    // Fetch studyrightsdata for student.
+    $data = Json::decode($this->fetchStudyRightsData($oodiId));
+    $date_today = date('Y-m-d', time());
+
+    // Make sure we have results to loop trough.
+    if(!$data) {
+      return null;
+    }
+
+    // Save all studyrights.
+    $studyrightprimalitychain = $data['data']['private_person']['studyRightPrimalityChain'];
+    $studyrights = $data['data']['private_person']['studyRights'];
+    $primarystudyright = $this->getPrimaryStudentDegreeProgram($oodiId);
+
+    $active_studyrights = [];
+    // Loop trough studyrights and save active studyrights.
+    foreach ($studyrights as $studyright) {
+      // Only save studyright if it's active ie. enddate null and startdate in the past
+      if($studyright['valid']['startDate'] < $date_today && (!$studyright['valid']['endDate'] || $studyright['valid']['endDate'] > $date_today)) {
+        // Handle specialization and graduation for a studyright
+        $studyrightdegreeprogram = $this->getActiveStudentDegreeProgram($studyright);
+
+        // Create new studyright
+        $studyrightdegree = new StudyRight($studyrightdegreeprogram);
+
+          // If primary, then set it so.
+        if($primarystudyright['id'] == $studyright['id']) {
+          $studyrightdegree.setPrimary(TRUE);
+        }
+
+        $active_studyrights[] = $studyrightdegree;
+      }
+    }
+
+    return $active_studyrights;
+  }
+
+  /**
+   * Get Student Primary Degree Program.
+   *
+   * @param int $oodiId
+   *   User Oodi ID.
+   *
+   * @return \Psr\Http\Message\ResponseInterface
+   *   Response object.
+   */
+  public function getPrimaryStudentDegreeProgram($oodiId) {
+    // Fetch studyrightsdata for student
+    $data = Json::decode($this->fetchStudyRightsData($oodiId));
+
+    if(!$data || $data['data']['private_person']) {
```

Comment:
I havent seen the real data source, but is this check for '$data['data']['private_person']' right?

Is a '!' possibly missing, almost same as using empty() here?